Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -118,6 +119,162 @@ Vector<String> calendar_fields(GlobalObject& global_object, Object& calendar, Ve
+        return {};
+    }
+
+    // 4. Return ? ToIntegerOrInfinity(result).
+    return result.to_integer_or_infinity(global_object);
+}
+
+// 12.1.10 CalendarMonth ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmonth
+double calendar_month(GlobalObject& global_object, Object& calendar, Object& date_like)
+{
+    auto& vm = global_object.vm();
+    // 1. Assert: Type(calendar) is Object.
+
+    // 2. Let result be ? Invoke(calendar, "month", « dateLike »).
+    auto result = calendar.invoke(vm.names.month, &date_like);
+    if (vm.exception())
+        return {};
+
+    // 3. If result is undefined, throw a RangeError exception.
+    if (result.is_undefined()) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidCalendar);
+        return {};
+    }
+
+    // 4. Return ? ToPositiveIntegerOrInfinity(result).
+    return to_positive_integer_or_infinity(global_object, result);
+}
+
+// 12.1.11 CalendarMonthCode ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmonthcode
+String calendar_month_code(GlobalObject& global_object, Object& calendar, Object& date_like)
+{
+    auto& vm = global_object.vm();
+    // 1. Assert: Type(calendar) is Object.
+
+    // 2. Let result be ? Invoke(calendar, "monthCode", « dateLike »).
+    auto result = calendar.invoke(vm.names.monthCode, &date_like);
+    if (vm.exception())
+        return {};
+
+    // 3. If result is undefined, throw a RangeError exception.
+    if (result.is_undefined()) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidCalendar);
+        return {};
+    }
+
+    // 4. Return ? ToString(result).
+    return result.to_string(global_object);
+}
+
+// 12.1.12 CalendarDay ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendarday
+double calendar_day(GlobalObject& global_object, Object& calendar, Object& date_like)
+{
+    auto& vm = global_object.vm();
+    // 1. Assert: Type(calendar) is Object.
+
+    // 2. Let result be ? Invoke(calendar, "day", « dateLike »).
+    auto result = calendar.invoke(vm.names.day, &date_like);
+    if (vm.exception())
+        return {};
+
+    // 3. If result is undefined, throw a RangeError exception.
+    if (result.is_undefined()) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidCalendar);
+        return {};
+    }
+
+    // 4. Return ? ToPositiveIntegerOrInfinity(result).
+    return to_positive_integer_or_infinity(global_object, result);
+}
+
+// 12.1.13 CalendarDayOfWeek ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendardayofweek
+Value calendar_day_of_week(GlobalObject& global_object, Object& calendar, Object& date_like)
+{
+    // 1. Assert: Type(calendar) is Object.
+
+    // 2. Return ? Invoke(calendar, "dayOfWeek", « dateLike »).
+    return calendar.invoke(global_object.vm().names.dayOfWeek, &date_like);
```

Comment:
I've gone from preferring the other style to actively disliking this one :sweat_smile:

would definitely prefer this, it reads so much nicer. where we get the VM from shouldn't matter.

```suggestion
    auto& vm = global_object.vm();
    ...
    // 2. Return ? Invoke(calendar, "dayOfWeek", « dateLike »).
    return calendar.invoke(vm.names.dayOfWeek, &date_like);
```

Reply:
switched it over to that :)